src/napari/__init__.py

```diff
@@ -13,6 +13,46 @@
 os.environ.setdefault('SPARSE_AUTO_DENSIFY', '1')
 limit_numpy1x_threads_on_macos_arm()
 
+
+def _check_installation_path():  # pragma: no cover
+    """Check for installation path conflicts.
+
+    Check if napari is present in site-packages. If napari is installed in editable mode,
+    notify the user of a the conflict that napari is also in site-packages.
+    """
+    import os
+    import sys
+    from pathlib import Path
+
+    if os.environ.get('CI', ''):
+        # Skip the check during testing
+        # we need to move to src layout to fix this
+        return
+    napari_installation_path = Path(__file__).absolute().parent.parent
+    if napari_installation_path.name == 'site-packages':
+        # napari is installed in non-editable mode
+        return
+
+    import numpy as np
+
+    site_packages_path = Path(np.__file__).absolute().parent.parent
+    if site_packages_path.name != 'site-packages':
+        # numpy is not installed in site-packages
+        return
+
+    problematic_napari_path = site_packages_path / 'napari'
+    if problematic_napari_path.exists():
+        print(  # noqa: T201
+            f'Found a napari directory: {problematic_napari_path}, '
+            'but napari is installed in editable mode. '
+            'Please remove napari directory from site-packages.',
+            file=sys.stderr,
+        )
+        raise RuntimeError('Mix of local and non local installation')
```

Comment:
```suggestion
    if napari_site_packages_path.exists():
        print(  # noqa: T201
            f'Found napari installed in a site-packages directory: {napari_site_packages_path}, '
            'but napari is also installed in editable mode. '
            'To prevent mixing local, editable and non-local, site-packages installation,
            ' please remove napari directory from site-packages.',
            file=sys.stderr,
        )
        raise RuntimeError('Conflict found where napari is present in local, editable location and non-local site-packages installation')
```